scene/gui/tab_container.cpp

```diff
@@ -883,85 +392,192 @@ void TabContainer::drop_data(const Point2 &p_point, const Variant &p_data) {
 			if (hover_now < 0) {
 				hover_now = get_tab_count() - 1;
 			}
-			move_child(get_tab_control(tab_from_id), get_tab_control(hover_now)->get_index());
+
+			move_child(get_tab_control(tab_from_id), get_tab_control(hover_now)->get_index(false));
 			set_current_tab(hover_now);
 		} else if (get_tabs_rearrange_group() != -1) {
-			// drag and drop between TabContainers
+			// Drag and drop between TabContainers.
 			Node *from_node = get_node(from_path);
 			TabContainer *from_tabc = Object::cast_to<TabContainer>(from_node);
 			if (from_tabc && from_tabc->get_tabs_rearrange_group() == get_tabs_rearrange_group()) {
 				Control *moving_tabc = from_tabc->get_tab_control(tab_from_id);
 				from_tabc->remove_child(moving_tabc);
-				add_child(moving_tabc, false, INTERNAL_MODE_FRONT);
+				add_child(moving_tabc);
```

Comment:
```suggestion
				add_child(moving_tabc, true);
```
This prevents weird names in case of conflicts:
![image](https://user-images.githubusercontent.com/2223172/156670022-137868cf-d424-444c-9367-0708ab7a54b8.png)